test/unit/Chainweb/Test/Pact5/RemotePactTest.hs

```diff
@@ -692,6 +695,104 @@ gasPurchaseFailureMessages rdb _step = runResourceT $ do
                 ? textContains "Failed to buy gas: Multiple gas payer capabilities"
 
+transitionOccurs :: RocksDb -> Step -> IO ()
+transitionOccurs rdb _step = runResourceT $ do
+    let v = instantCpmTransitionTestVersion petersonChainGraph
+    --let v = pact5InstantCpmTestVersion petersonChainGraph
+    let cid = unsafeChainId 0
+    fx <- mkFixture v rdb
+
+    liftIO $ do
+        checkPactVersion fx v cid >>= P.equals Pact4
+        forM_ @_ @_ @Word [1..17] $ \i -> do
+            advanceAllChains_ fx
+            -- index trick to show which iteration fails, if any
+            (i,) <$> checkPactVersion fx v cid >>= P.equals (i, Pact4)
+        advanceAllChains_ fx
+        checkPactVersion fx v cid >>= P.equals Pact5
+
+-- | Test that xchains work across the Pact4->Pact4 transition boundary.
+--   This is mostly the same as 'spvTest', except it waits for the transition.
+transitionCrosschain :: RocksDb -> Step -> IO ()
+transitionCrosschain rdb step = runResourceT $ do
+    let v = instantCpmTransitionTestVersion petersonChainGraph
+    let srcChain = unsafeChainId 0
+    let targetChain = unsafeChainId 9
+    fx <- mkFixture v rdb
+
+    let checkIsVersion pv = do
+            checkPactVersion fx v srcChain >>= P.equals pv
+            checkPactVersion fx v targetChain >>= P.equals pv
+
+    liftIO $ do
+        checkIsVersion Pact4
+
+        initiator <- buildTextCmd v
+            $ set cbSigners
+                [ mkEd25519Signer' sender00
+                    [ CapToken (QualifiedName "GAS" (ModuleName "coin" Nothing)) []
+                    , CapToken (QualifiedName "TRANSFER_XCHAIN" (ModuleName "coin" Nothing))
+                        [ PString "sender00"
+                        , PString "sender01"
+                        , PDecimal 1.0
+                        , PString (chainIdToText targetChain)
+                        ]
+                    ]
+                ]
+            $ set cbRPC (mkExec ("(coin.transfer-crosschain \"sender00\" \"sender01\" (read-keyset 'k) \"" <> chainIdToText targetChain <> "\" 1.0)") (mkKeySetData "k" [sender01]))
+            $ defaultCmd srcChain
+
+        step "xchain initiate"
+        send fx v srcChain [initiator]
+        let initiatorReqKey = cmdToRequestKey initiator
+        (sendCut, _) <- advanceAllChains fx
+        [Just sendCr] <- pollWithDepth fx v srcChain [initiatorReqKey] (Just (ConfirmationDepth 0))
+        let cont = fromMaybe (error "missing continuation") (_crContinuation sendCr)
+
+        step "waiting until pact5 transition"
+
+        step "... performing transition"
+        replicateM_ 16 $ advanceAllChains_ fx
+        checkIsVersion Pact4
+        advanceAllChains_ fx
+        checkIsVersion Pact5
+
+        let sendHeight = sendCut ^?! ixg srcChain . blockHeight
+        spvProof <- createTransactionOutputProof_ (fx ^. to _cutFixture . CutFixture.fixtureWebBlockHeaderDb) (fx ^. to _cutFixture . CutFixture.fixturePayloadDb) targetChain srcChain sendHeight 0
```

Comment:
I just realized we aren't using the `/spv` endpoint here or in the original SPV test, maybe we should start doing that in a subsequent PR just for more realism.